src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -54,18 +54,16 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PropertyGroup>
       <DefaultComposeDir>$(HOME)</DefaultComposeDir>
       <DefaultComposeDir Condition="$(OS.ToUpper().Contains('WINDOWS'))">$(USERPROFILE)</DefaultComposeDir>
-      <DefaultComposeDir>$([System.IO.Path]::Combine($(DefaultComposeDir), '.dotnet', 'packages'))</DefaultComposeDir>
+      <DefaultComposeDir>$([System.IO.Path]::Combine($(DefaultComposeDir), '.dotnet', $(PlatformTarget), 'packages'))</DefaultComposeDir>
+      <ComposeDir Condition="'$(ComposeDir)' != '' and '$(DoNotDecorateComposeDir)' == ''">$([System.IO.Path]::Combine($(ComposeDir), $(PlatformTarget)))</ComposeDir>
       <ComposeDir Condition="'$(ComposeDir)' == ''">$(DefaultComposeDir)</ComposeDir>
-      <ComposeDir Condition="'$(DoNotDecorateComposeDir)' == ''">$(ComposeDir)$(PlatformTarget)\$(_TFM)\</ComposeDir>
+      <ComposeDir Condition="'$(DoNotDecorateComposeDir)' == ''">$([System.IO.Path]::Combine($(ComposeDir), $(_TFM)))</ComposeDir>
```

Comment:
This condition seems wrong IMO.  What if I set `<DoNotDecorateComposeDir>false</DoNotDecorateComposeDir>`?